crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -140,6 +128,15 @@ impl Acir {
         let mut output = match &ins.operation {
             Operation::Binary(binary) => self.evaluate_binary(binary, ins.res_type, evaluator, ctx),
             Operation::Not(_) => todo!(),
+            Operation::Constrain(value) => {
+                let mut value = self.substitute(*value, evaluator, ctx);
+                if value.witness.is_none() {
+                    value.witness = Some(generate_witness(&value, evaluator));
+                }
+                let eq_zero = from_witness(evaluate_zero_equality(&value, evaluator));
+                // != 0
+                subtract(&Arithmetic::one(), FieldElement::one(), &eq_zero).into()
```

Comment:
@guipublic could you take a look at this?

Reply:
Will do after the merge of #221, I don't want to merge again and again.

Reply:
Lets not hold up other PRs waiting for others to merge since its unclear how long until #221 will be merged itself. I can help with that merge as well if you'd like